File: src/main/java/com/banktech/javachallenge/service/domain/submarine/SubmarineResponse.java
package com.banktech.javachallenge.service.domain.submarine;

import com.banktech.javachallenge.service.domain.game.SimpleResponse;

import java.util.List;

public class SubmarineResponse extends SimpleResponse {
    private List<OwnSubmarine> submarines;

    public SubmarineResponse() {
    }

    public List<OwnSubmarine> getSubmarines() {
        return submarines;
    }

    public void setSubmarines(List<OwnSubmarine> submarines) {
        this.submarines = submarines;
    }

    @Override
    public String toString() {
        return "SubmarineResponse{" +
                "submarines=" + submarines +
                '}';
    }
}

File: src/main/java/com/banktech/javachallenge/service/domain/submarine/SubmarineResponse.kt
package com.banktech.javachallenge.service.domain.submarine

import com.banktech.javachallenge.service.domain.Position

/**
 *
 */
data class SubmarineResponse(val type: String,
                             val id: Long,
                             val position: Position,
                             val owner: Owner,
                             val velocity: Int,
                             val angle: Double,
                             val hp: Int,
                             val sonarCooldown: Int,
                             val torpedoCooldown: Int,
                             val sonarExtended: Int)

fun SubmarineResponse.toSubmarine(): Submarine {
    return Submarine(type, id, position, owner, velocity, angle)
}